### PR TITLE
docs: added python helps you quickly with Apache APISIX plugin development blog

### PR DESCRIPTION
**Pull request by shuaijinchao**
Changes:

added python helps you quickly with Apache APISIX plugin development blog

### CODE REVIEW

**Comment by liuxiran** [comment]
cc @Serendipity96 @yzeng25 to have a look, thanks

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+authorURL: "https://github.com/shuaijinchao"
+authorImageURL: "https://avatars.githubusercontent.com/u/8529452?v=4"
+keywords:
+
+- Python
+- APISIX
+- APISIX Python 插件
+- APISIX Python plugin
+- apisix-python-plugin-runner
+- APISIX Python Plugin Runner tags: [technology]
```
Comment (liuxiran) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]:
```suggestion
- APISIX Python Plugin Runner 
tags: [technology]
```

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+- `filter` 函数参数只能包含 `Request` 和 `Response` 类对象作为参数
+- `Request` 对象参数可以获取请求信息
+- `Response` 对象参数可以设置响应信息
+- `self.config` 可以获取插件配置信息
+- `filter` 函数中调用 `self.stop()` 时将马上中断请求，响应数据。
+- `filter` 函数中调用 `self.rewrite()` 时，将会继续请求。
+
+## 欢迎参与
+
+目前 `Apache APISIX` 各语言的 `Runner` 还处于早期开发阶段，我们会陆续完善其功能。成功的开源项目离不开大家的参与和贡献，欢迎各位参与到 `Apache APISIX Runner`
```
Comment (juzhiyuan) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]:
```suggestion
目前 `Apache APISIX` 各语言的 `Runner` 还处于早期开发阶段，我们会陆续完善其功能。成功的开源项目离不开大家的参与和贡献，欢迎各位参与到 `Apache APISIX Runner` 的开发中来，让我们一起共建 `Apache APISIX` 和各语言的桥梁。
```

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+的开发中来，让我们一起共建 `Apache APISIX` 和各语言的桥梁。
```
Comment (juzhiyuan) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]:
```suggestion
```

**Review by moonming — APPROVED** [review]

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
+---
+title: "Python 助你快速上手 Apache APISIX 插件开发"
+author: "shuaijinchao"
+authorURL: "https://github.com/shuaijinchao"
+authorImageURL: "https://avatars.githubusercontent.com/u/8529452?v=4"
+keywords:
+
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: please delete empty line, thanks.

Comment (Serendipity96) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: extra space

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+<!--truncate-->
+
+在 `Apache APISIX Python Runner` 之前社区中已经支持了 `Java` 和 `Golang` 语言的 `Runner` ，今天 `Python Runner`
+也来了，社区中的小伙伴们在开发 `Apache APISIX` 的插件时又多了一种新选择。
+
+## 简介
+
+### Apache APISIX
+
+`APISIX` 是一个高性能的云原生开源API网关，它可以对请求进行统一的拦截和治理（如：`鉴权`、`认证`、`缓存`、`版本`、`熔断`、`审计`
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: `APISIX` ---> `Apache APISIX`

Comment (yzeng25) [reply website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: "`鉴权`、`认证`、`缓存`、`版本`、`熔断`、`审计`" ---> "鉴权、认证、缓存、版本、熔断、审计"

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+- APISIX Python plugin
+- apisix-python-plugin-runner
+- APISIX Python Plugin Runner 
+tags: [technology]
+
+---
+
+<!--truncate-->
+
+在 `Apache APISIX Python Runner` 之前社区中已经支持了 `Java` 和 `Golang` 语言的 `Runner` ，今天 `Python Runner`
```
Comment (Serendipity96) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: Would you like to add 'Java' and 'Go' articles?
https://apisix.apache.org/blog/2021/06/21/use-Java-to-write-Apache-APISIX-plugins/
https://apisix.apache.org/blog/2021/08/19/go-makes-Apache-APISIX-better/

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+### Apache APISIX
+
+`APISIX` 是一个高性能的云原生开源API网关，它可以对请求进行统一的拦截和治理（如：`鉴权`、`认证`、`缓存`、`版本`、`熔断`、`审计`
+等等）帮助开发人员轻松的对外提供安全可靠的服务，而开发人员通过 `Apache APISIX` 的加持只需要关注业务实现即可，省去了大量花费在通用能力上的开发与维护上的时间并且也降低了整体业务架构的复杂度。
+
+### Python
+
+`Python` 语言作为一个解释型的高级编程语言，它 `语法简洁易上手`、`代码可读性好` ，在 `跨平台` 、`可移植性` 、`开发效率`
+上都有很好的表现，同时作为一个高级编程语言它的封装抽象程度比较高屏蔽了很多底层细节（例如：`GC`
+）让我们在开发的过程中可以更专注应用逻辑的开发，而且作为一个有30年历史的老牌开发语言生态以及各种模块已经非常完善，我们大部分的开发和应用场景都可以从社区中找到很成熟的模块或解决方案，`Python`
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: "......让我们在开发的过程中可以更专注应用逻辑的开发，而且作为一个有30年历史的老牌开发语言生态以及各种模块已经非常完善，"
--->
"让我们在开发的过程中可以更专注应用逻辑的开发。Python 作为一个有30年历史的老牌开发语言，它的生态以及各种模块已经非常完善。"

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+其他的优点就不在一一赘述，`Python` 的缺点也比较明显作为一门解释性语言相较于 `C++` 、`Golang` 这样的编译型语言性能上的差距还是比较大的。
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: "`Python`其他的优点就不在一一赘述，"
--->
"`Python`的其他优点就不再一一赘述。"

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md (hunk 5 above)
Comment (Serendipity96) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: APISIX  -> Apache APISIX

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+多语言插件的执行顺序是在 `ext-plugin-*` 插件配置项中定义的，像其他插件一样，它们可以被启用并在运行中重新定义。
+
+## 部署测试
+
+### 基础运行环境
+
+- Apache APISIX 2.7
+- Python 3.6+
+
+`APISIX` 的安装部署本文不在过多赘述可以参照 [文档](https://github.com/apache/apisix/blob/master/docs/en/latest/how-to-build.md) 进行部署。
```
Comment (Serendipity96) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: APISIX -> Apache APISIX

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md (hunk 8 above)
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: “`Python` 的缺点也比较明显作为一门解释性语言相较于 `C++` 、`Golang` 这样的编译型语言性能上的差距还是比较大的。”
--->
“`Python` 的缺点也比较明显：`Python` 作为一门解释性语言，相较于 `C++` 和 `Golang` 这样的编译型语言，在性能上的差距还是比较大的。”

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md (hunk 6 above)
Comment (Serendipity96) [reply website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: I think it's easier for others who want to know how to use 'Java' or 'Go' to find the article by link.
在 `Apache APISIX Python Runner` 之前社区中已经支持了 [Java](https://xxx) 和 [Golang](https://xxx) 语言的 `Runner`

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+`Python` 语言作为一个解释型的高级编程语言，它 `语法简洁易上手`、`代码可读性好` ，在 `跨平台` 、`可移植性` 、`开发效率`
+上都有很好的表现，同时作为一个高级编程语言它的封装抽象程度比较高屏蔽了很多底层细节（例如：`GC`
+）让我们在开发的过程中可以更专注应用逻辑的开发，而且作为一个有30年历史的老牌开发语言生态以及各种模块已经非常完善，我们大部分的开发和应用场景都可以从社区中找到很成熟的模块或解决方案，`Python`
+其他的优点就不在一一赘述，`Python` 的缺点也比较明显作为一门解释性语言相较于 `C++` 、`Golang` 这样的编译型语言性能上的差距还是比较大的。
+
+### Apache APISIX Python Runner
+
+[apache-apisix-python-runner](https://github.com/apache/apisix-python-plugin-runner) 这个项目可以理解为 `Apache APISIX`
+和 `Python`
+之间的一道桥梁，通过 `Python Runner` 可以把 `Python` 直接应用到 `APISIX` 的插件开发中，最重要的还是希望让更多对 `Apache APISIX` 和 API网关 感兴趣的 `Pythonista`
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: API网关 ---> API 网关

Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: Is there a way to emphasize `Pythonista` in Chinese? When I google search this word, it returns a bunch of ambiguous results, https://www.google.com/search?q=Pythonista&oq=Pythonista&aqs=chrome..69i57.270j0j7&sourceid=chrome&ie=UTF-8 .

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+
+上图左边是 `Apache APISIX` 的工作流程，右边的 `Plugin Runner` 是各语言的插件运行器，本文介绍的 `apisix-python-plugin-runner` 就是支持 `Python`
+语言的 `Plugin Runner`。
+
+当你在 `Apache APISIX` 中配置一个 `Plugin Runner` 时，`Apache APISIX` 会启动一个子进程运行 `Plugin Runner`，该子进程与 `Apache APISIX`
+进程属于同一个用户，当我们重启或重新加载 `Apache APISIX` 时，`Plugin Runner` 也将被重启。
+
+如果你为一个给定的路由配置了 `ext-plugin-*` 插件，请求命中该路由时将触发 `Apache APISIX` 通过 `Unix Socket` 向 `Plugin Runner` 发起 `RPC` 调用。调用分为两个阶段：
+
+- `ext-plugin-pre-req` : 在执行 `Apache APISIX` 内置插件(Lua 语言插件)之前
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: "`ext-plugin-pre-req` : " ---> "`ext-plugin-pre-req`："

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+- `ext-plugin-post-req` : 在执行 `Apache APISIX` 内置插件(Lua 语言插件)之后
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: ditto

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+
+根据需要配置 `Plugin Runner` 的执行时机。
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: I don't get it... could you explain more in this sentence?

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md (hunk 9 above)
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: "`APISIX` 的安装部署本文不在过多赘述可以参照 [文档](https://github.com/apache/apisix/blob/master/docs/en/latest/how-to-build.md) 进行部署。"
--->
"`Apache APISIX` 的安装部署不再过多赘述，详情请参考 [Apache APISIX 官方文档：如何构建 Apache APISIX](https://github.com/apache/apisix/blob/master/docs/zh/latest/how-to-build.md) 进行部署。"

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 14]
```diff
@@ -0,0 +1,275 @@
...
+$ cd /path/to/apisix
+# Start or Restart
+$ ./bin/apisix [ start | restart ]
+```
+
+启动或重启 `APISIX` 即可，此时 `APISIX` 和 `Python Runner` 已经完成配置并启动。
+
+### 测试 Python Runner
+
+#### 配置APISIX路由及插件信息
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: 配置APISIX路由及插件信息  ---> 配置 Apache APISIX 路由及插件信息

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md (hunk 11 above)
Comment (Serendipity96) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: Is this a plugin? Please add a link to the project documentation or GitHub address.

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md (hunk 12 above)
Comment (Serendipity96) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: Is this a plugin? Please add a link to the project documentation or GitHub address.

**Comment by shuaijinchao** [comment]
cc @Serendipity96 @yzeng25 @juzhiyuan @liuxiran fixed, please review

File: website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md [hunk 15]
```diff
@@ -0,0 +1,279 @@
...
+
+在 `Apache APISIX Python Runner` 之前社区中已经支持了 [Java](https://github.com/apache/apisix-java-plugin-runner)
+和 [Go](https://github.com/apache/apisix-go-plugin-runner) 语言的 `Runner` ，今天 `Python Runner`
+也来了，社区中的小伙伴们在开发 `Apache APISIX` 的插件时又多了一种新选择。
+
+## 简介
+
+### Apache APISIX
+
+`Apache APISIX` 是一个高性能的云原生开源API网关，它可以对请求进行统一的拦截和治理（如：鉴权、认证、缓存、版本、熔断、审计 等等）帮助开发人员轻松的对外提供安全可靠的服务，而开发人员通过 `Apache APISIX`
```
Comment (yzeng25) [inline website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: ”`Apache APISIX` 是一个高性能的云原生开源API网关“
--->
"`Apache APISIX` 是一个高性能的云原生开源 API 网关"

Comment (yzeng25) [reply website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: Other things LGTM @shuaijinchao cc @Serendipity96 @juzhiyuan @liuxiran

Comment (shuaijinchao) [reply website/blog/2021-09-06-python-helps-you-quickly-with-Apache-APISIX-development.md]: done

**Review by spacewander — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Pull request merged**